File: churn-demo/new/account.py
class Account:
    def __init__(self, owner, balance):
        self.owner = owner
        self.balance = balance
        self.min_balance = 0.0
        self.history = []

    def deposit(self, amount):
        self.balance = self.balance + amount
        self.history.append("DEPOSIT")

    def withdraw(self, amount):
        self.balance = self.balance - amount
        self.history.append("WITHDRAW")

    def apply_fee(self, fee):
        self.balance = self.balance - fee
        self.history.append("FEE")

    def get_balance(self):
        return self.balance

    def is_overdrawn(self):
        return self.balance < self.min_balance

    def history_count(self):
        return len(self.history)

    def apply_interest(self, rate):
        self.balance = self.balance + self.balance * rate
        self.history.append("INTEREST")

File: churn-demo/new/Account.java
import java.util.ArrayList;
import java.util.List;

public class Account {
    private String owner;
    private double balance;
    private double minBalance = 0.0;
    private List<String> history = new ArrayList<>();

    public Account(String owner, double balance) {
        this.owner = owner;
        this.balance = balance;
    }

    public void deposit(double amount) {
        balance = balance + amount;
        history.add("DEPOSIT");
    }

    public void withdraw(double amount) {
        balance = balance - amount;
        history.add("WITHDRAW");
    }

    public void applyFee(double fee) {
        balance = balance - fee;
        history.add("FEE");
    }

    public double getBalance() {
        return balance;
    }

    public boolean isOverdrawn() {
        return balance < minBalance;
    }

    public int historyCount() {
        return history.size();
    }

    public void applyInterest(double rate) {
        balance = balance + balance * rate;
        history.add("INTEREST");
    }

    public boolean transfer(Account to, double amount) {
        balance = balance - amount;
        to.balance = to.balance + amount;
        return true;
    }
}
